File: NgCmsWeb/src/app/models/route.model.ts
import { Serializable } from './../interfaces/serializable.interface';

export class RouteModel implements Serializable<RouteModel> {
    guid: string;
    path: string;
    parentRouteGuid: string;

    deserialize(input) {
        this.guid = input.guid;
        this.path = input.path;
        this.parentRouteGuid = input.parentRouteGuid;

        return this;
    }
}

File: NgCmsApi/Models/RouteModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace NgCmsApi.Models
{
    public class RouteModel
    {
        [Required]
        [JsonProperty("guid")]
        public Guid Guid { get; set; }

        [Required]
        [JsonProperty("path")]
        public string Path { get; set; }

        [JsonProperty("parentRouteGuid")]
        public Guid? ParentRouteGuid { get; set; }
    }
}
